src/pages/integrations/gift-cards/confirm-invoice/confirm-invoice.ts

```diff
@@ -0,0 +1,385 @@
+    navParams: NavParams,
+    private incomingDataProvider: IncomingDataProvider,
+    onGoingProcessProvider: OnGoingProcessProvider,
+    popupProvider: PopupProvider,
+    profileProvider: ProfileProvider,
+    txConfirmNotificationProvider: TxConfirmNotificationProvider,
+    txFormatProvider: TxFormatProvider,
+    walletProvider: WalletProvider,
+    translate: TranslateService,
+    public payproProvider: PayproProvider,
+    platformProvider: PlatformProvider,
+    walletTabsProvider: WalletTabsProvider,
+    clipboardProvider: ClipboardProvider,
+    events: Events,
+    AppProvider: AppProvider
+  ) {
+    super(
+      actionSheetProvider,
+      bwcErrorProvider,
+      bwcProvider,
+      configProvider,
+      decimalPipe,
+      feeProvider,
+      giftCardProvider,
+      replaceParametersProvider,
+      externalLinkProvider,
+      logger,
+      modalCtrl,
+      navCtrl,
+      navParams,
+      onGoingProcessProvider,
+      popupProvider,
+      profileProvider,
+      txConfirmNotificationProvider,
+      txFormatProvider,
+      walletProvider,
+      translate,
+      payproProvider,
+      platformProvider,
+      walletTabsProvider,
+      clipboardProvider,
+      events,
+      AppProvider
+    );
+    this.hideSlideButton = false;
+    this.invoiceName = this.navParams.data.invoiceName;
+    this.configWallet = this.configProvider.get().wallet;
+  }
+
+  async ngOnInit() {
+    this.invoiceData = this.navParams.data.invoiceData;
+    this.invoiceId = this.navParams.data.invoiceId;
+    this.amount = this.invoiceData.price || 1;
+    this.currency = this.invoiceData.currency || 'USD';
+    this.onlyIntegers = this.invoiceData.currency === 'JPY';
+    this.merchantProvidedEmail = this.navParams.data.email;
+    let { emailAddress } = this.invoiceData.buyerProvidedInfo;
+    this.buyerProvidedEmail = emailAddress;
+    this.email = this.merchantProvidedEmail
+      ? this.merchantProvidedEmail
+      : this.buyerProvidedEmail
+      ? this.buyerProvidedEmail
+      : await this.getEmail();
+  }
+
+  ionViewDidLoad() {
+    this.logger.info('Loaded: ConfirmInvoicePage');
+  }
+
+  ionViewWillEnter() {
+    this.isOpenSelector = false;
+    this.navCtrl.swipeBackEnabled = false;
+    const { BITPAY_API_URL } = this.invoiceProvider.credentials;
+    this.network = this.invoiceProvider.getNetwork();
+    this.wallets = this.profileProvider.getWallets({
+      onlyComplete: true,
+      network: this.network,
+      hasFunds: true
+    });
+    this.invoiceUrl = `${BITPAY_API_URL}/invoice/${this.invoiceId}`;
+    const { selectedTransactionCurrency } = this.invoiceData.buyerProvidedInfo;
+    if (selectedTransactionCurrency) {
+      this.wallets = _.filter(this.wallets, (x: any) => {
+        return (
+          x.credentials.coin == selectedTransactionCurrency.toLowerCase() &&
+          !this.profileProvider.vaultHasWallet(x.credentials.walletId)
+        );
+      });
+    }
+    this.wallet = this.wallets[0];
```

Comment:
Sets first wallet as this.wallet by default. Most users only have a few wallets.